File: optionals/med-4.0.0/tests/c/2.3.6/test6.c
/*  This file is part of MED.
 *
 *  COPYRIGHT (C) 1999 - 2019  EDF R&D, CEA/DEN
 *  MED is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  MED is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public License
 *  along with MED.  If not, see <http://www.gnu.org/licenses/>.
 */


/******************************************************************************
 * - Nom du fichier : test6.c
 *
 * - Description : ecriture d'elements dans un maillage MED
 *
 *****************************************************************************/

#include <med.h>
#define MESGERR 1
#include <med_utils.h>

#ifdef DEF_LECT_ECR
#define MODE_ACCES MED_LECTURE_ECRITURE
#elif DEF_LECT_AJOUT
#define MODE_ACCES MED_LECTURE_AJOUT
#else
#define MODE_ACCES MED_CREATION
#endif

int main (int argc, char **argv)


{
  med_idt fid;
  med_int nse2 = 5;
  med_int se2[10] = {1,2,1,3,2,4,3,4,2,3};
  /*                                12345678901234561234567890123456123456789012345612345678901234561234567890123456*/
  char nomse2[MED_TAILLE_PNOM*5+1]="se1             se2             se3             se4             se5             ";
  med_int numse2[5] = {1,2,3,4,5};
  med_int nufase2[5] = {-1,-1,0,-2,-3}; 
  med_int ntr3 = 2;
  med_int tr3[6] = {1,2,-5,-5,3,-4};
  /*                                  12345678901234561234567890123456*/
  char nomtr3[MED_TAILLE_PNOM*2+1] = "tr1             tr2             ";
  med_int numtr3[2] = {4,5};
  med_int nufatr3[2] = {0,-1};
  char maa[MED_TAILLE_NOM+1] = "maa1";
  med_int mdim = 2;

  /* ouverture du fichier */
  if ( (fid = MEDouvrir("test6.med",MODE_ACCES)) < 0) {
    MESSAGE("Impossible de creer le fichier test6.med : ");
    return -1;
  }
  
  /* creation du maillage */
  if ( MEDmaaCr(fid,maa,mdim,MED_NON_STRUCTURE,
		   "un maillage pour test6") < 0 ) {
    MESSAGE("Impossible de creer le maillage : ");
    return -1;
  }
  
  /* ecriture des connectivites des segments */
  if ( MEDconnEcr(fid,maa,mdim,se2,MED_FULL_INTERLACE,nse2,MED_ARETE,MED_SEG2,
		  MED_DESC) < 0 ) {
    MESSAGE("Impossible d'ecrire la connectivité des segments: ");
    return -1;
  }


  /* ecriture (optionnelle) des noms des segments */
  if ( MEDnomEcr(fid,maa,nomse2,nse2,MED_ARETE,MED_SEG2) < 0 ) {
    MESSAGE("Impossible d'ecrire les noms des segments : ");
    return -1;
  }
  
  /* ecriture (optionnelle) des numeros des segments */
  if ( MEDnumEcr(fid,maa,numse2,nse2,MED_ARETE,MED_SEG2) < 0 ) {
    MESSAGE("Impossible d'ecrire les numéros des segments : ");
    return -1;
  }
  
    /* ecriture des numeros des familles des segments */
  if ( MEDfamEcr(fid,maa,nufase2,nse2,MED_ARETE,MED_SEG2) < 0 ) {
    MESSAGE("Impossible d'ecrire les numéros de famille des segments : ");
    return -1;
  }

  /* ecriture des connectivites des triangles */
  if ( MEDconnEcr(fid,maa,mdim,tr3,MED_FULL_INTERLACE,ntr3,MED_MAILLE,MED_TRIA3,
		     MED_DESC) < 0 ) {
    MESSAGE("Impossible d'ecrire la connectivité des triangles : ");
    return -1;
  }

  /* ecriture (optionnelle) des noms des triangles */
  if ( MEDnomEcr(fid, maa, nomtr3, ntr3, MED_MAILLE, MED_TRIA3) < 0 ) {
    MESSAGE("Impossible d'ecrire les noms des triangles : ");
    return -1;
  }
  
  /* ecriture (optionnelle) des numeros des triangles */
  if ( MEDnumEcr(fid,maa,numtr3,ntr3,MED_MAILLE,MED_TRIA3) < 0 ) {
    MESSAGE("Impossible d'ecrire les numéros des triangles : ");
    return -1;
  }

  /* ecriture des numeros des familles des triangles */
  if ( MEDfamEcr(fid,maa,nufatr3,ntr3,MED_MAILLE,MED_TRIA3) < 0 ) {
    MESSAGE("Impossible d'ecrire les numéros de famille des triangles : ");
    return -1;
  }

  /* fermeture du fichier */
  if ( MEDfermer(fid) < 0 ) {
    MESSAGE("Impossible de fermerle fichier : ");
    return -1;
  }

  return 0;
 
}

File: optionals/med-4.0.0/tests/f/2.3.6/test6.f
C*  This file is part of MED.
C*
C*  COPYRIGHT (C) 1999 - 2019  EDF R&D, CEA/DEN
C*  MED is free software: you can redistribute it and/or modify
C*  it under the terms of the GNU Lesser General Public License as published by
C*  the Free Software Foundation, either version 3 of the License, or
C*  (at your option) any later version.
C*
C*  MED is distributed in the hope that it will be useful,
C*  but WITHOUT ANY WARRANTY; without even the implied warranty of
C*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
C*  GNU Lesser General Public License for more details.
C*
C*  You should have received a copy of the GNU Lesser General Public License
C*  along with MED.  If not, see <http://www.gnu.org/licenses/>.
C*

C       *******************************************************************************
C       * - Nom du fichier : test6.f
C       *
C       * - Description : exemples d'ecriture d'elements dans un maillage MED
C       *
C       ******************************************************************************
	program test6
C       
	implicit none
	include 'med.hf'
C
C	
	integer*8 fid
	integer cret
	
	integer     mdim,nse2,ntr3
	parameter  (nse2 = 5, ntr3 = 2, mdim = 2)
	integer     se2 (2*nse2)
	character*16 nomse2(nse2)
	integer     numse2(nse2),nufase2(nse2)

	integer     tr3 (3*ntr3)
	character*16 nomtr3(ntr3)
	integer     numtr3(ntr3), nufatr3(ntr3) 
	character*32 maa 

	data se2     / 1,2,1,3,2,4,3,4,2,3 /    
	data nomse2  /"se1","se2","se3","se4","se5" / 
	data numse2  / 1,2,3,4,5 /, nufase2 /-1,-1,0,-2,-3/
	data tr3     /1,2,-5,-5,3,-4 /, nomtr3 /"tr1","tr2"/,
     &                                  numtr3 /4,5/
	data nufatr3 /0,-1/,  maa /"maa1"/

C       ** Ouverture du fichier                            **
 	call efouvr(fid,'test6.med',MED_LECTURE_ECRITURE, cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur creation du fichier'
	   call efexit(-1)
	endif      

C       ** Creation du maillage maa de dimension 2         **
	call efmaac(fid,maa,mdim,MED_NON_STRUCTURE,
     &                 'un maillage pour test6',cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur creation du maillage'
	   call efexit(-1)
	endif      

C       ** Ecriture des connectivites des segments         **
	call efcone(fid,maa,mdim,se2,MED_NO_INTERLACE,
     &               nse2,MED_ARETE,
     &               MED_SEG2,MED_DESC,cret ) 
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture de la connectivite'
	   call efexit(-1)
	endif      

C       ** Ecriture (optionnelle) des noms des segments    **
	call efnome(fid,maa,nomse2,nse2,MED_ARETE,
     &	                      MED_SEG2 ,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture des noms'
	   call efexit(-1)
	endif      

C       ** Ecriture (optionnelle) des numeros des segments **
	call efnume(fid,maa,numse2,nse2,
     &              MED_ARETE ,MED_SEG2,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture des numeros'
	   call efexit(-1)
	endif      

C	** Ecriture des numeros des familles des segments  **
	call effame(fid,maa,nufase2,nse2,
     &              MED_ARETE,MED_SEG2,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture des numéros de famille'
	   call efexit(-1)
	endif      

C       ** Ecriture des connectivites des triangles        **
	call efcone(fid,maa,mdim,tr3,MED_NO_INTERLACE,
     &              ntr3,MED_MAILLE,
     &	            MED_TRIA3,MED_DESC,cret ) 
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture de la connectivite'
	   call efexit(-1)
	endif      

C       ** Ecriture (optionnelle) des noms des triangles   **
	call efnome(fid,maa,nomtr3,ntr3,MED_MAILLE,
     &	                      MED_TRIA3,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture des noms'
	   call efexit(-1)
	endif      

C       ** Ecriture (optionnelle) des numeros des triangles **
	call efnume(fid,maa,numtr3,ntr3,MED_MAILLE,
     &                       MED_TRIA3,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture des numeros'
	   call efexit(-1)
	endif      

C      ** Ecriture des numeros des familles des triangles  **
	call effame(fid,maa,nufatr3,ntr3,MED_MAILLE,
     &                      MED_TRIA3,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur ecriture des numeros de famille'
	   call efexit(-1)
	endif      

C       ** Fermeture du fichier   **
	call efferm (fid,cret)
	print *,cret
	if (cret .ne. 0 ) then
	   print *,'Erreur a la fermeture du fichier'
	   call efexit(-1)
	endif      
C
	end
